standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1802,12 +1802,12 @@ private MTable convertToMTable(Table tbl) throws InvalidObjectException,
     // accordingly
     String tableType = tbl.getTableType();
     boolean isExternal = Boolean.parseBoolean(tbl.getParameters().get("EXTERNAL"));
-    if (TableType.MANAGED_TABLE.toString().equals(tableType)) {
+    if (TableType.MANAGED_TABLE.toString().equalsIgnoreCase(tableType)) {
```

Comment:
I think, this issue doesn't exist in master code.
We use Boolean.parseBoolean to read the "EXTERNAL" property which ignores case. Also, the fix in the patch is to use equalsIgnoreCase when compare the TableType and not "EXTERNAL" property which is irrelevant.
If I misunderstood your scenario, please let me know.
Also, it would be better to add a unit test to reproduce this bug in master code.

Reply:
@sankarh The table type gets converted as expected but the describe formatted view shows it wrongly.
Example: I converted the below Managed table to External using ` .. SET TBLPROPERTIES ('EXTERNAL'='True')`
![image](https://user-images.githubusercontent.com/6907950/53644278-3546c900-3c5c-11e9-9f26-aceaed0eed7b.png)

If I do the same using `.. SET TBLPROPERTIES ('EXTERNAL'='TRUE')`
![image](https://user-images.githubusercontent.com/6907950/53644443-a25a5e80-3c5c-11e9-8171-f7fb0b3eea41.png)

Let me know if you need further details.

Reply:
@animenon 
I tested it myself and it worked fine for me. Probably, you are using different version of Hive or Java.
Take a look at this code which converts managed to external type.
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java#L4092
It parses using Boolean.parseBoolean which ignores case and hence converts to EXTERNAL_TABLE type.

Reply:
Even, if it behaves differently for different cases of EXTERNAL property, still I'm not sure how your fix would solve the issue as you are ignoring case of TableType but not EXTERNAL property. Pls explain.

Reply:
IgnoreCase used [here](https://github.com/apache/hive/blob/e4d60e29910ec709d432fd85819cb6b6d097d689/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1810) :
`if (TableType.EXTERNAL_TABLE.toString().equalsIgnoreCase(tableType))`
 for EXTERNAL property and tableType equals check would check if equals ignoring the case of letters. 

I agree the tableType  ignoreCase [here](https://github.com/apache/hive/blob/e4d60e29910ec709d432fd85819cb6b6d097d689/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1805) may not be required.

Reply:
Yes the `Boolean.parseBoolean` would ignore case but the thing is you wouldn't reach there.. 
`if (TableType.EXTERNAL_TABLE.toString().equalsIgnoreCase(tableType))` is checked before `if (!isExternal)`
[ref](https://github.com/apache/hive/blob/e4d60e29910ec709d432fd85819cb6b6d097d689/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1811)

Like you pointed out rightly, `isExternal` ignores case but the flow wouldn't goto that condition because the condition before that fails.

Reply:
@animenon 
Thanks for thr reply! But, I'm still confused. 
As per your point, isExternal is proper (both "True" and "TRUE"). But, why would tableType have different case based on input for EXTERNAL property?
Below is the code that reads EXTERNAL property from ALTER TABLE command and sets correct tableType.
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java#L4092
      ```
String externalProp = alterTbl.getProps().get("EXTERNAL");
      if (externalProp != null) {
        if (Boolean.parseBoolean(externalProp) && tbl.getTableType() == TableType.MANAGED_TABLE) {
          tbl.setTableType(TableType.EXTERNAL_TABLE);
        } else if (!Boolean.parseBoolean(externalProp) && tbl.getTableType() == TableType.EXTERNAL_TABLE) {
          tbl.setTableType(TableType.MANAGED_TABLE);
        }
      }
```
If you notice, we use Boolean.parseBoolean(externalProp) here which ignores case.
I didn't get how two values of externalProp ("True" and "TRUE") influences the table type set via setTableType. 
Pls point to what I'm missing here. 

@rmsmani 
Please check if I miss anything.

Reply:
@sankarh 
How about changing the condition to equals in the below lines (both if & else condition)
        if (Boolean.parseBoolean(externalProp) && tbl.getTableType() == TableType.MANAGED_TABLE) {

Reply:
@rmsmani and @animenon 
HIVE-16324 and HIVE-19753 have already fixed this bug in apache/master. So, this bug doesn't exist anymore. 
Probably, you might see it in older Hive versions which doesn't include these patches. 
So, please close this jira as invalid.

Reply:
@sankarh - For the newer version, HIVE-16324 seems to fix it, the version of hive I am using still has this issue, will check my hive version update in appropriate branch. 
Thanks for checking on this.